src/coreclr/interpreter/compiler.cpp

```diff
@@ -3863,13 +3863,32 @@ void InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
                 int32_t token = getI4LittleEndian(m_ip + 1);
                 void *str;
                 InfoAccessType accessType = m_compHnd->constructStringLiteral(m_compScopeHnd, token, &str);
-                DeclarePointerIsString(str);
```

Comment:
You might consider keeping this bit if accessType is IAT_VALUE, it makes looking ath the disasm output much more useful.

Reply:
I kept it, I just moved it further down I think. Will double-check.